service/storage_proxy.cc

```diff
@@ -445,7 +445,7 @@ class abstract_write_response_handler : public seastar::enable_shared_from_this<
         _stats.background_writes++;
         _proxy->_global_stats.background_write_bytes += _mutation_holder->size();
         _throttled = false;
-        _ready.set_value();
+        _ready.set_value(bo::success());
```

Comment:
The default constructor for `result<>` is explicitly deleted. The author of the boost outcome library explains this decision in the FAQ: https://www.boost.org/doc/libs/1_71_0/libs/outcome/doc/html/faq.html#why-is-implicit-default-construction-disabled

Reply:
Sad. Yet another benefit to not using this library :-(
It sounds like the issue is that result<> also allows holding "nothing" - not a result and not an exception. I think this is a mistake (unless I missing something here). result<T> should hold T or an exception, not nothing. Moving a result<T> should be like moving T - and leave behind whatever moving T moves behind.

Reply:
> It sounds like the issue is that result<> also allows holding "nothing" - not a result and not an exception.

This was an issue with boost outcome v1's result<>. We are using v2, which does not have an empty state (from the FAQ: "v2 has eliminated any legal possibility of being empty"). I think this implies that the move constructor should work in the way you would like it to work.

Reply:
So I think these guys made a mistake in their judgement. Not only does it make sense for result<T>() to call T's default constructor, it's even more obvious this should be allowed when T is nothing (which I think is the case here) :-(
It's weird that we need to include that bizarre namespace just to spell out this "success()" instead of nothing.
But I guess it's not something we can fix now (if we use boost::outcome).